Fails to compile with GHC-8.2.1-rc3

Using https://gist.github.com/tfausak/623d283cfe7a338bcbfca56ddfec0ecd I get the following error:

```
    /run/user/499/stack2779/log-warper-1.1.2/src/System/Wlog/CanLog.hs:70:13: error:
        • The default type signature for dispatchMessage:
            forall (t :: (* -> *) -> * -> *) (n :: * -> *).
            (MonadTrans t, t n ~ m, CanLog n) =>
            LoggerName -> Severity -> Text -> t n ()
          does not match its corresponding non-default type signature
        • When checking the class method:
            dispatchMessage :: forall (m :: * -> *).
                               CanLog m =>
                               LoggerName -> Severity -> Text -> m ()
          In the class declaration for ‘CanLog’
       |
    70 |     default dispatchMessage :: (MonadTrans t, t n ~ m, CanLog n)
       |             ^^^^^^^^^^^^^^^
```

cc @ChShersh @volhovM

--- a/src/System/Wlog/CanLog.hs
+++ b/src/System/Wlog/CanLog.hs
@@ -71,7 +71,7 @@ class Monad m => CanLog m where
                             => LoggerName
                             -> Severity
                             -> Text
-                            -> t n ()
+                            -> m ()
     dispatchMessage name sev t = lift $ dispatchMessage name sev t
 
 instance CanLog IO where

